config/proxy.go

```diff
@@ -32,6 +33,24 @@ func (c *Config) setProxy() error {
 		if err != nil {
 			return err
 		}
+
+		if c.HTTPBackend.CertFile != "" && c.HTTPBackend.KeyFile != "" {
+			readCert, err := tls.LoadX509KeyPair(
+				c.HTTPBackend.CertFile,
+				c.HTTPBackend.KeyFile,
+			)
+			if err != nil {
+				return err
+			}
+
+			config := &tls.Config{
+				Certificates: []tls.Certificate{readCert},
+			}
+
+			tr := &http.Transport{TLSClientConfig: config}
+			httpClient = &http.Client{Transport: tr}
```

Comment:
Could we do `httpClient.Transport = tr` here instead of creating a new `HttpClient`?